yt/data_objects/tests/test_data_containers.py

```diff
@@ -0,0 +1,121 @@
+from yt.utilities.exceptions import YTFieldNotFound
+
+
+def test_yt_data_container():
+    # Test if ds could be None
+    with assert_raises(RuntimeError) as err:
+        YTDataContainer(None, None)
+    desired = 'Error: ds must be set either through class type or parameter' \
+              ' to the constructor'
+    assert_equal(str(err.exception), desired)
+
+    # Test if field_data key exists
+    ds = fake_random_ds(5)
+    proj = ds.proj("density", 0, data_source=ds.all_data())
+    assert_equal('px' in proj.keys(), True)
+    assert_equal('pz' in proj.keys(), False)
+
+    # Delete the key and check if exits
+    proj.__delitem__('px')
+    assert_equal('px' in proj.keys(), False)
+    proj.__delitem__('density')
+    assert_equal('density' in proj.keys(), False)
+
+    # Delete a non-existent field
+    with assert_raises(YTFieldNotFound) as ex:
+        proj.__delitem__('p_mass')
+    desired = "Could not find field '('stream', 'p_mass')' in UniformGridData."
+    assert_equal(str(ex.exception), desired)
+
+    # Test the convert method
+    assert_equal(proj.convert('HydroMethod'), -1)
+
+def test_write_out():
+    filename = "sphere.txt"
+    ds = fake_particle_ds()
+    sp = ds.sphere(ds.domain_center, 0.25)
+    sp.write_out(filename)
+
+    with open(filename, "r") as file:
+        file_row_1 = file.readline()
+        file_row_2 = file.readline()
+        file_row_2 = np.array(file_row_2.split('\t'), dtype=np.float64)
+    sorted_keys = sorted(sp.field_data.keys())
+    _keys = [str(k) for k in sorted_keys]
+    _keys = "\t".join(["#"] + _keys + ["\n"])
+    _data = [sp.field_data[k][0] for k in sorted_keys]
+
+    assert_equal(_keys, file_row_1)
+    assert_array_equal(_data, file_row_2)
+
+def test_save_object():
+    ds = fake_particle_ds()
+    sp = ds.sphere(ds.domain_center, 0.25)
+    sp.save_object("my_sphere_1", filename="test_save_obj")
+    obj = shelve.open("test_save_obj", protocol=-1)
+    loaded_sphere = obj["my_sphere_1"][1]
+    assert_array_equal(loaded_sphere.center, sp.center)
+    assert_equal(loaded_sphere.radius, sp.radius)
+    obj.close()
+
+    sp.save_object("my_sphere_2")
+
+def test_to_dataframe():
+    try:
+        import pandas as pd
+        pd # to ignore Flake8 error
+        fields = ["density", "velocity_z"]
+        ds = fake_random_ds(6)
+        dd = ds.all_data()
+        df1 = dd.to_dataframe(fields)
+        assert_array_equal(dd[fields[0]], df1[fields[0]])
+        assert_array_equal(dd[fields[1]], df1[fields[1]])
+    except ImportError:
+        pass
+
+def test_std():
+    ds = fake_random_ds(3)
+    ds.all_data().std('density', weight="velocity_z")
+
+def test_to_frb():
+    ds = fake_amr_ds(fields=["density", "cell_mass"], geometry="cylindrical",
+                     particles=16**3)
+    proj = ds.proj("density", weight_field="cell_mass", axis=1,
+                   data_source=ds.all_data())
+    proj.to_frb((1.0, 'unitary'), 64)
```

Comment:
Can you test that the `FixedResolutionBuffer` object you get back is correct? (e.g. it produces images with a resolution of `(64, 64)` and that the image covers the full domain? For the latter you can just inspect the state of the `FixedResolutionBuffer` object (e.g. by checking `frb.bounds`) rather than trying to make sure the image is correct.